main.tf

```diff
@@ -1,81 +1,205 @@
-# Set our cloud provider and access details
+################################################################################
+# Set our cloud provider and access details,
+# such as AWS security credentials
+# https://console.aws.amazon.com/iam/home?region=us-east-1#/security_credentials
+# it stored in ./terraform.tdvars
+#
+################################################################################
 provider "aws" {
   access_key = var.access_key
   secret_key = var.secret_key
   region     = var.region
 }
 
-# Create a VPC to launch our instances into
+data "aws_availability_zones" "available" {}
+################################################################################
+# Create a VPC to launch main VPC-network
+# determine the CIDR block for our VPC
+#
+#
+# https://www.terraform.io/docs/providers/aws/d/vpc.html
+################################################################################
 resource "aws_vpc" "main-vpc" {
-  cidr_block        = "172.20.0.0/16"
-  security_group_id = "${aws_security_group.front.id}"
+  cidr_block           = var.cidr["main"]
+  enable_dns_hostnames = true
+  enable_dns_support   = "true"
```

Comment:
Is it necessary to use double quotes here?

Reply:
althought no, you can access each element using the dot-separated attribute notation, but:

Map/object attributes with names that are valid identifiers can also be accessed using the dot-separated attribute notation, like local.object.attrname. In cases where a map might contain arbitrary user-specified keys, we recommend using only the square-bracket index notation (local.map["keyname"]).

Reply:
yep, it is fixed)